app/uk/gov/hmrc/ngrpropertylinkingfrontend/controllers/ConnectionToPropertyController.scala

```diff
@@ -0,0 +1,110 @@
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class ConnectionToPropertyController @Inject()(connectionToPropertyView: ConnectionToPropertyView,
+                                               authenticate: AuthRetrievals,
+                                               isRegisteredCheck: RegistrationAction,
+                                               mcc: MessagesControllerComponents,
+                                               propertyLinkingRepo: PropertyLinkingRepo)(implicit appConfig: AppConfig, ec: ExecutionContext)
+  extends FrontendController(mcc) with I18nSupport {
+
+  private val ownerButton: NGRRadioButtons = NGRRadioButtons("Owner", Owner, Some("Owns the property."))
+  private val occupierButton: NGRRadioButtons = NGRRadioButtons("Occupier",Occupier, Some("Operates from the property."))
+  private val bothButton: NGRRadioButtons = NGRRadioButtons("Owner and occupier",OwnerAndOccupier, Some("Owns and Operates from the property."))
+  private val ngrRadio: NGRRadio = NGRRadio(NGRRadioName("connection-to-property-radio"), Seq(ownerButton, occupierButton, bothButton))
+
+  def show: Action[AnyContent] = {
+    (authenticate andThen isRegisteredCheck).async { implicit request =>
+      propertyLinkingRepo.findByCredId(CredId(request.credId.getOrElse(""))).flatMap {
+        case Some(properties) =>
+          val propertyAddress = properties.vmvProperty.addressFull
+          Future.successful(Ok(connectionToPropertyView(
+            form = form(),
+            radios = buildRadios(form(), ngrRadio),
+            navigationBarContent = createDefaultNavBar(),
+            propertyAddress = propertyAddress)))
+        case None => Future.failed(throw new NotFoundException("Unable to find matching postcode"))
+      }
+    }
+  }
+
+  def submit: Action[AnyContent] =
+    (authenticate andThen isRegisteredCheck).async { implicit request =>
+      form().bindFromRequest().fold(
+        formWithErrors => {
+          val credId = request.credId.getOrElse("")
+          propertyLinkingRepo.findByCredId(CredId(credId)).flatMap {
+            case Some(properties) =>
+              val propertyAddress = properties.vmvProperty.addressFull
+              Future.successful(BadRequest(connectionToPropertyView(
+                form = formWithErrors,
+                radios = buildRadios(formWithErrors, ngrRadio),
+                navigationBarContent = createDefaultNavBar(),
+                propertyAddress = propertyAddress
+              )))
+            case None =>
+              Future.successful(Redirect(routes.NoResultsFoundController.show))
+          }
+        },
+        {
+          case ConnectionToPropertyForm.Owner =>
+            propertyLinkingRepo.insertConnectionToProperty(
+              credId = CredId(request.credId.getOrElse("")),
+              connectionToProperty = "Owner"
```

Comment:
So we have a lot of repeated strings lurking around this repo and thos strings are being used as constants. I think it would be better to have a Location where 

val owner: String = "Owner"

is set and then the rest of the code user Location.owner rather than "Owner"